Sources/SwiftSyntax/LexerError.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// If the token has a lexical error, this defines the type of the error.
+/// `lexerErrorOffset` in the token will specify at which offset the error
+/// occurred.
+public enum LexerError {
```

Comment:
Any thoughts on something like:
```
public struct LexerError {
  public enum Kind {
    case ...
  }

  public let kind: Kind
  public let byteOffset: UInt16 // No need for alias, it's only defined in one place now

  public init(...) { ... }
}
```

And then just having the one `public var lexerError: LexerError` field and a
```
public var hasLexerError: Bool {
  return lexerError != nil
}
```
?

This would also simplify all the comments that mention "if there's an error, `lexerErrorOffset` will specify the offset".

As an aside, I'd personally be fine with `UInt8` as the offset type. Do people really have more than 255 characters for a token?

Reply:
That’s a much better idea. No idea why I modeled it the way I did…

> As an aside, I'd personally be fine with UInt8 as the offset type. Do people really have more than 255 characters for a token?

String literals are currently single tokens and those can reasonably be bigger than 255 characters, I think.

Reply:
Oh, unfortunate.